dna/tests/src/project/project_init.rs

```diff
@@ -0,0 +1,121 @@
+            }),
+        );
+        let header_hash = fixt!(HeaderHash);
+        mock_hdk
+            .expect_create()
+            .with(mockall::predicate::eq(expected))
+            .times(1)
+            .return_const(Ok(header_hash.clone()));
+        
+        // init also calls join_project_during_init, drilling down, it calls the following hdk functions under the hood
+        // TODO: test when the MEMBER_PATH doesn't exist in DHT
+        let member_path = Path::from("member");
+        let member_path_entry = Entry::try_from(member_path).unwrap();
+        let member_path_entry_hash = fixt!(EntryHash);
+        mock_hdk
+            .expect_hash_entry() // called from `Path::from(MEMBER_PATH).ensure()?;`
+            .with(mockall::predicate::eq(member_path_entry.clone()))
+            .times(1)
+            .return_const(Ok(member_path_entry_hash.clone()));
+
+
+        let member_path_get_input = vec![GetInput::new(
+            AnyDhtHash::from(member_path_entry_hash.clone()), GetOptions::content()
+        )];
+        // assuming the path exists on DHT
+        let expected_get_output = vec![Some(fixt!(Element))];
+        mock_hdk
+            .expect_get() // called from `Path::from(MEMBER_PATH).ensure()?;`
+            .with(mockall::predicate::eq(member_path_get_input))
+            .times(1)
+            .return_const(Ok(expected_get_output));
+
+        mock_hdk
+            .expect_hash_entry() // called from `let member_path_address = Path::from(MEMBER_PATH).hash()?;`
+            .with(mockall::predicate::eq(member_path_entry.clone()))
+            .times(1)
+            .return_const(Ok(member_path_entry_hash.clone()));
+        
+        let agent_info = fixt!(AgentInfo);
+        mock_hdk
+            .expect_agent_info()
+            .times(1)
+            .return_const(Ok(agent_info.clone()));
+        
+        let member = Member {
+            address: WrappedAgentPubKey(agent_info.agent_initial_pubkey),
+        };
+        let create_member_input = EntryWithDefId::try_from(member.clone()).unwrap();
+        mock_hdk
+            .expect_create()
+            .with(mockall::predicate::eq(create_member_input))
+            .times(1)
+            .return_const(Ok(header_hash.clone()));
+        
+        let member_hash = fixt!(EntryHash);
+        let member_entry = Entry::try_from(member.clone()).unwrap();
+        mock_hdk
+            .expect_hash_entry() // called from `let member_entry_hash = hash_entry(&member)?;`
+            .with(mockall::predicate::eq(member_entry))
+            .times(1)
+            .return_const(Ok(member_hash.clone()));
+        let create_link_input = CreateLinkInput::new(
+            member_path_entry_hash,
+            member_hash,
+            LinkTag::from(()),
+        );
+        mock_hdk
+            .expect_create_link()
+            .with(mockall::predicate::eq(create_link_input))
+            .times(1)
+            .return_const(Ok(header_hash));
```

Comment:
do the same with this one.

Then, within the main test, call those two functions, passing in the `mock_hdk`